modules/imgproc/src/shapedescr.cpp

```diff
@@ -471,7 +471,7 @@ static RotatedRect fitEllipseNoDirect( InputArray _points )
 
     // store angle and radii
     rp[4] = -0.5 * atan2(gfp[2], gfp[1] - gfp[0]); // convert from APP angle usage
-    if( fabs(gfp[2]) > min_eps )
+    if (fabs(gfp[2]) / fabs(gfp[1] - gfp[0]) > min_eps)
```

Comment:
Could you please replace division operation with multiplication (micro optimization) and left comment with original condition?

```
// fabs(gfp[2]) / fabs(gfp[1] - gfp[0]) > min_eps
if (fabs(gfp[2]) > fabs(gfp[1] - gfp[0]) * min_eps)
```